### DISCUSSION TITLE
Second level of archive

### DISCUSSION THREAD

**Discussion by laurentkempe** [statiqdev/Discussions#134] [Q&A] [answered]
I am trying to build something in which I generate grouped archives per year and per month, [like the following](https://laurentkempe.com/all-archives/)

![image](https://user-images.githubusercontent.com/272612/155078395-34ab8bd7-e0fc-4a03-9aa6-35bc9fef1272.png)

I can click on the year to see all blog posts for that year. If I click on the month, I will see all blog posts for that month of that year.
I could easily do the first level, but I don't find a way to do the month's second level and would appreciate any idea.

This is the way I do the year, and which creates my index as I want it

    ArchiveSources: => $"posts/**/*"
    ArchiveKey: => GetDateTime("date").Year
    ArchivePageSize: => 10
    ArchiveDestination: >
      => GetInt("Index") <= 1 ? $"archives/{GetString("GroupKey")}/index.html" : $"archives/{GetString("GroupKey")}/page/{GetInt("Index")}/index.html"

**Comment by TheBrambleShark** [comment]
To set you off in the right direction, one thing you could look at is string interpolation. Anything after the `=>` can be C#

`$"{GetDateTime("date").Year}-{GetDateTime("date").Month}"` could be a possible ArchiveKey value.

You would likely have to change your ArchiveDestination key too, but I feel this may get you closer to what you're looking for.

Also, I don't really know the api of `GetDateTime()` that well, so if there's a way you can get the year and month in one step, I would do that. For instance, if it returns a `DateTime`, you could do this instead, using [DateTime Format Strings](https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings):
```cs
$"{GetDateTime("date"): yyyy-MM}" // 2022-02
```

I'm sure you could also make that `2022/02` as well, which would be nice for having pages like `2022/index.html`, `2022/02/index.html`, and `2022/02/some-post.html`

**Reply by laurentkempe** [reply]
Thanks @Foxtrek64 I did not think of using a combination of year and month for the ArchiveKey 👍🏼
I went with 

    ArchiveKey: => $"{GetDateTime("date").Year}/{GetDateTime("date").Month:00}" // 2022/02

Then I did not need to change my ArchiveDestination

    ArchiveDestination: >
      => GetInt("Index") <= 1 ? $"archives/{GetString("GroupKey")}/index.html" : $"archives/{GetString("GroupKey")}/page/{GetInt("Index")}/index.html"

**Comment by daveaglick** [comment, marked as answer]
I keep meaning to work up an example and keep running out of time. I'll still do that and post it here, but in the meantime I think the key to what you want is two archives. Do one archive for month-by-month (and include the year in the key as @Foxtrek64 mentioned) and then another archive for year-by-year. Then in the archive list template for the year-by-year archive, group the archive documents by month and add a link to where you know the month-by-month archive summary will be written.

Does that make sense?

**Reply by laurentkempe** [reply]
Thanks @daveaglick it makes sense! The trick I missed was to have two archives and combined with ArchiveKey of @Foxtrek64 I have a solution working.

Just asking myself now. With two archives, I need two templates. One using the year as ArchiveKey will render the index page for year and month (like on screenshot), and all archives index pages per year. How could I have the second with ArchiveKey rendering all the archives per month, but not rendering any index file?
